### PR TITLE
Make release template more consise and generic

### PR DESCRIPTION
**Pull request by Anviking**
# Issue Number

None. Release.

# Overview

- [x] The template was out of date. I started with the latest actually-used template
- [x] I generalised tags with `{{VARIABLE_NAME}}` syntax
- [x] I removed "main features" section (redundant with API & CLI docs)
- [x] I removed the "artefact table" (redundant with the GH-generated list)

# Comments

- Sorry, I should have used two commits to differentiate between the most-recently-used version and my tweaks. But cf overview.
- In action: https://github.com/input-output-hk/cardano-wallet/releases

### CODE REVIEW

**Comment by KtorZ** [comment]
cc @tatyanavych , I think this is something we should discuss with you :point_up:

File: .github/RELEASE_TEMPLATE.md [hunk 1]
```diff
@@ -1,150 +1,95 @@
...
 
 <!-- Specific installation steps for this particular release. This should
 basically captures whatever is currently available on the repository at
-the moment of releasing. --> 
+the moment of releasing. -->
 
 ## Linux 64-bit
 
-1. Install [jormungandr@X.Y.Z](https://github.com/input-output-hk/jormungandr/releases/tag/vX.Y.Z) from the official repository.
+1. Install [jormungandr@{{JORM_TAG}}](https://github.com/input-output-hk/jormungandr/releases/tag/{{JORM_TAG}}) from the official repository.
```
Comment (piotr-iohk) [inline .github/RELEASE_TEMPLATE.md]: Perhaps we should also mention cardano-node here? 
```
1. Install jormungandr@VERSION or cardano-node@VERSION
2. Download respective cardano-wallet binary (jormungandr or cardano-node compatible)
```
... and maybe add similar info for all: Linux, Windows, MacOS and Docker...

**Review by tatyanavych — COMMENTED** [review]
Great - it is more concise, and looks more as a release notes. Release Notes should enumerate changes that happened since the last release. Plus relevant docs, and system requirements, and artifacts. 
* OK to remove Section title OVERVIEW, but it is *desirable* to provide an overview of a release in the first statement.
* CHANGELOG should be internal file that may be referenced from Release Notes (e.g. in Overview statement or after enumeration of all changes as a ref for more details). It may have many details that end user may not wish to read.
* The recommended sequence of sections in release notes (wording may vary)
1) New Features
2) Changed features || Improvements
3) Discontinued Features || Removed Features
4) Breaking API changes 
5) Resolved Issues || Bug fixes
6) Known Issues
...
Sections have to be present only of there are relevant changes. 
See recommendations [here](https://www.prodpad.com/blog/writing-release-notes/), example in [cardano-node](https://github.com/input-output-hk/cardano-node/releases/tag/1.9.0) and Release Notes Template in Confuence on Releases site

File: .github/RELEASE_TEMPLATE.md (hunk 1 above)
Comment (Anviking) [reply .github/RELEASE_TEMPLATE.md]: 61fc85f74d41a73add840514096514f95e852522

**Comment by tatyanavych** [comment]
_CHANGELOG should be moved down e.g. before **Weekly Reports**.
After brief first overview statement (if you are going to write it) there should be_ 

**New Features**
feature1
feature2
...
**Improvements**
imp1
imp2
...
**Bug Fixes**
bf1
bf2
....
**Known Issues**
ki1
ki2
...
For more details see CHANGELOG
**CHANGELOG**

**Review by tatyanavych — DISMISSED** [review]
Remove `v` from
`CARDANO_NODE_TAG="v1.9.3"`
to have
`CARDANO_NODE_TAG="1.9.3"` 
so that the URL would be
https://github.com/input-output-hk/cardano-node/releases/tag/1.9.3

**Comment by Anviking** [comment]
Oops, indeed, thanks! 👍 

(looking at following https://github.com/input-output-hk/cardano-wallet/pull/1533#issuecomment-607918047 too)

**Comment by Anviking** [comment]
Ok, I have edited this [specific draft release](https://github.com/input-output-hk/cardano-wallet/releases), will change the template in this PR later.

- Added a new features-, and improvements- sections (@KtorZ could you check that the content is ok?)
- Reorganization / made headlines smaller matching cardano-node
- Fixed broken cardano-node link / tag

cc @tatyanavych @piotr-iohk

File: scripts/release-notes.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+echo "$BYRON_CLI_WIKI_COMMIT (Byron CLI)"
+
+rm -rf $TMP_WIKI_DIR
+cd $TMP_CWD
+
+# Release-specific parameters (Change when you bump the version)
+GIT_TAG="v2020-04-01"
+CABAL_VERSION="2020.4.1"
+JORM_TAG="v0.8.15"
+CARDANO_NODE_TAG="v1.9.3"
```
Comment (piotr-iohk) [inline scripts/release-notes.sh]: Small thing, but we need to drop v (off course it'll be edited every time, but just for the reference)

File: scripts/release-notes.sh [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+CHANGELOG=GENERATED_CHANGELOG.md
```
Comment (piotr-iohk) [inline scripts/release-notes.sh]: Perhaps we could add a script for generating a changelog also (or maybe include those steps in this script)?

**Review by piotr-iohk — APPROVED** [review]
LGTM! (I suppose modulo Tatyana's suggestions about the [structure](https://github.com/input-output-hk/cardano-wallet/pull/1533#issuecomment-607918047) and minor comments)
Really great that you automated this!

File: .github/RELEASE_TEMPLATE.md [hunk 4]
```diff
@@ -1,150 +1,89 @@
-# Overview
+Compatible with [`jormungandr@{{JORM_TAG}}`](https://github.com/input-output-hk/jormungandr/releases/tag/{{JORM_TAG}}) and [`cardano-node@{{CARDANO_NODE_TAG}}`](https://github.com/input-output-hk/cardano-node/releases/tag/{{CARDANO_NODE_TAG}}).
```
Comment (KtorZ) [inline .github/RELEASE_TEMPLATE.md:3]: This one feels a bit redundant with the installation instruction below. I'd suggest moving the tags into the title below to make them standing out a bit.

File: .github/RELEASE_TEMPLATE.md [hunk 5]
```diff
@@ -1,150 +1,89 @@
...
+<!-- Specific installation steps for this particular release. This should
+basically captures whatever is currently available on the repository at
+the moment of releasing. -->
 
-# Key Features
+## Jormungandr / ITN
 
-<!-- What's included in the software, mention what's new in this particular
-release. Use: ![][new] -->
+1. Install [jormungandr@{{JORM_TAG}}](https://github.com/input-output-hk/jormungandr/releases/tag/{{JORM_TAG}}).
```
Comment (KtorZ) [inline .github/RELEASE_TEMPLATE.md]: neat-picking --> the link should be in back quotes, like for Byron below.

File: .github/RELEASE_TEMPLATE.md [hunk 6]
```diff
@@ -1,150 +1,89 @@
...
+2. Download the provided `cardano-wallet-jormungandr` for your platform, and uncompress it in a directory that is on your `$PATH`, e.g. `/usr/local/bin`. Or `%PATH%` on Windows.
 
-## Server
+3. (optional) Install the bash/zsh auto-completion script according to the [jormungandr cli manual](https://github.com/input-output-hk/cardano-wallet/wiki/Wallet-Command-Line-Interface/{{JORM_CLI_WIKI_COMMIT}})
 
-## Internal
+4. Start `cardano-wallet --help` and see available parameters.
 
-## Known Limitations
+## Byron / Mainnet
```
Comment (KtorZ) [inline .github/RELEASE_TEMPLATE.md]: The target component really is `cardano-node` though, not `Byron`. What about the following formulation:

## cardano-node / Byron

File: .github/RELEASE_TEMPLATE.md [hunk 7]
```diff
@@ -1,150 +1,89 @@
...
 ---                            | ---                 | ---:
-Matthias Benkort @KtorZ        | Technical Team Lead | :hourglass:
-Piotr Stachyra @piotr-iohk     | QA Engineer         | :hourglass:
-Tatyana Valkevych @tatyanavych | Release Manager     | :hourglass:
+Matthias Benkort @KtorZ        | Technical Team Lead | :hourglass: 
+Piotr Stachyra @piotr-iohk     | QA Engineer         | :hourglass: 
+Tatyana Valkevych @tatyanavych | Release Manager     | :hourglass: 
 
-[new]: https://raw.githubusercontent.com/input-output-hk/cardano-wallet/master/.github/images/badge-new.png 
+[new]: https://raw.githubusercontent.com/input-output-hk/cardano-wallet/master/.github/images/badge-new.png
```
Comment (KtorZ) [inline .github/RELEASE_TEMPLATE.md]: I guess this badge is no longer useful since the "key features" section was dropped.

**Review by KtorZ — APPROVED** [review]
Looks good. One question though, why did you remove the "TroubleShooting" section and the bits about libcrypto? 
Was this moved to the wiki?

File: scripts/release-notes.sh (hunk 3 above)
Comment (Anviking) [reply scripts/release-notes.sh]: Yes. Hopefully that works out well. That and and the `.cabal` `.nix` version-bumping.

File: .github/RELEASE_TEMPLATE.md [hunk 8]
```diff
@@ -1,150 +1,89 @@
...
-  $ mv $HOME/.local/bin/cardano-wallet.sh > /etc/bash_completion.d/cardano-wallet.sh
-  $ source /etc/bash_completion.d/cardano-wallet.sh
-  ```
-
-4. Start `cardano-wallet --help` and see available parameters.
-
-#### TroubleShooting
-
-<details>
-  <summary>cardano-wallet: error while loading shared libraries: <code>libcrypto.so.1.0.0</code>: cannot open shared object file: No such file or directory</summary>
```
Comment (Anviking) [inline .github/RELEASE_TEMPLATE.md:44]: > One question though, why did you remove the "TroubleShooting" section and the bits about libcrypto?
Was this moved to the wiki?

@KtorZ This section was in the template, but I don't think it was ever used in an actual release. I started with replacing the template with the most recently used release-notes.

I guess we could add it?

File: .github/RELEASE_TEMPLATE.md (hunk 6 above)
Comment (Anviking) [reply .github/RELEASE_TEMPLATE.md]: Done.

File: .github/RELEASE_TEMPLATE.md (hunk 4 above)
Comment (Anviking) [reply .github/RELEASE_TEMPLATE.md:3]: It is a bit redundant, but I believed the compatible node versions were important enough to be shown here at the top.

You might be actively seeking a cardano-wallet release based on the node version it supports.

File: .github/RELEASE_TEMPLATE.md (hunk 8 above)
Comment (Anviking) [reply .github/RELEASE_TEMPLATE.md:44]: Re-added.

File: .github/RELEASE_TEMPLATE.md [hunk 9]
```diff
@@ -1,150 +1,114 @@
...
 
-## Windows 64-bit
+<!-- A snapshot of the documentation at the time of releasing. -->
 
-1. Install [jormungandr@X.Y.Z](https://github.com/input-output-hk/jormungandr/releases/tag/vX.Y.Z) from the official repository.
+| Link                                                                                                                                        | Audience                                                   |
+| ---                                                                                                                                         | ---                                                        |
+| [API Documentation](https://input-output-hk.github.io/cardano-wallet/api/{{GIT_TAG}})                                                       | Users of the Cardano Wallet API                            |
+| CLI Manual: [jormungandr](https://github.com/input-output-hk/cardano-wallet/wiki/Wallet-Command-Line-Interface/{{JORM_CLI_WIKI_COMMIT}}) / [byron](https://github.com/input-output-hk/cardano-wallet/wiki/Wallet-Command-Line-Interface-(cardano-wallet-byron)/{{BYRON_CLI_WIKI_COMMIT}}) | Users of the Cardano Wallet API                            |
```
Comment (KtorZ) [inline .github/RELEASE_TEMPLATE.md]: This one should also be updated to be more consistent with the naming, it's either ITN vs Byron or, jormungandr vs cardano-node. 

Also, it feels kinda weird to speak about the `jormungandr cli manual` for the cardano-wallet command-line. What about something like `w/ jormungandr` and `w/ cardano-node` and then refer to them as `cardano-wallet w/ jormungandr cli manual` & `cardano-wallet w/ cardano-node cli manual` ?

Comment (Anviking) [reply .github/RELEASE_TEMPLATE.md]: I'd go with ` ITN / Byron` then because it fits on a single line.
